src/Workspaces/Core/Portable/Recommendations/AbstractRecommendationService.cs

```diff
@@ -29,6 +29,26 @@ internal abstract class AbstractRecommendationService : IRecommendationService
             return symbols;
         }
 
+        protected static IEnumerable<ISymbol> GetRecommendedNamespaceNameSymbols(
+            SemanticModel semanticModel, SyntaxNode declarationSyntax, CancellationToken cancellationToken)
+        {
+            var containingNamespaceSymbol = semanticModel.Compilation.GetCompilationNamespace(
+                semanticModel.GetEnclosingNamespace(declarationSyntax.SpanStart, cancellationToken));
+
+            var symbols = semanticModel.LookupNamespacesAndTypes(declarationSyntax.SpanStart, containingNamespaceSymbol)
+                                       .Where(recommendationSymbol => IsNonIntersectingNamespace(recommendationSymbol, declarationSyntax));
+
+            return symbols;
+        }
+
+        protected static bool IsNonIntersectingNamespace(ISymbol recommendationSymbol, SyntaxNode declarationSyntax)
+        {
+            return recommendationSymbol.IsNamespace() &&
```

Comment:
It just filters out the declaration the user is in the middle of typing, and for VB it filters out any namespaces nested within an existing declaration (since they can never be peers of the declaration).

I'll add the comment.